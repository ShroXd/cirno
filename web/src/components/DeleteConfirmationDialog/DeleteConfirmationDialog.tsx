import { useTranslation } from 'react-i18next'

import {
  Button,
  Dialog,
  DialogBody,
  DialogFooter,
  DialogHeader,
} from '@material-tailwind/react'

import { useDelete } from '~/hooks/useDelete'

interface DeleteConfirmationDialogProps {
  mediaLibraryId: number
  title: string
  description: string
  open: boolean
  handleOpen: () => void
}

export const DeleteConfirmationDialog = ({
  mediaLibraryId,
  title,
  description,
  open,
  handleOpen,
}: DeleteConfirmationDialogProps) => {
  const { t } = useTranslation()
  const del = useDelete()

  const handleDelete = async () => {
    console.log('delete media library', mediaLibraryId)
    const res = await del(`/library/${mediaLibraryId}`)
    console.log('res', res)
    handleOpen()
  }

  return (
    <Dialog open={open} handler={handleOpen} size='xs'>
      <DialogHeader>{title}</DialogHeader>
      <DialogBody>{description}</DialogBody>
      <DialogFooter className='flex justify-end gap-4'>
        <Button variant='text' onClick={handleOpen}>
          {t('common.cancel')}
        </Button>
        <Button color='red' onClick={handleDelete}>
          {t('common.confirm')}
        </Button>
      </DialogFooter>
    </Dialog>
  )
}
