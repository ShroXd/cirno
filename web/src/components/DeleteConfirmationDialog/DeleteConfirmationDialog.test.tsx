import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { DeleteConfirmationDialog } from './DeleteConfirmationDialog'

describe('DeleteConfirmationDialog', () => {
  it('should render without a11y violations', async () => {
    const { container } = render(
      <DeleteConfirmationDialog
        title='Delete Confirmation'
        description='Are you sure you want to delete this item?'
        open={true}
        handleConfirm={() => {}}
        handleCancel={() => {}}
      />
    )
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly with passed props', () => {
    const { baseElement } = render(
      <DeleteConfirmationDialog
        title='Delete Confirmation'
        description='Are you sure you want to delete this item?'
        open={true}
        handleConfirm={() => {}}
        handleCancel={() => {}}
      />
    )

    expect(baseElement).toBeVisible()
    expect(baseElement).toMatchSnapshot()
  })
})
