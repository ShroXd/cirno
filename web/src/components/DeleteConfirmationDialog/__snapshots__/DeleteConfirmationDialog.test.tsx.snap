// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteConfirmationDialog > should render correctly with passed props 1`] = `
<body
  data-floating-ui-scroll-lock=""
  style="overflow: hidden; padding-right: 1024px;"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-floating-ui-portal=""
    id=":r3:"
  >
    <div
      style="position: fixed; overflow: auto; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 9999;"
    >
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="grid place-items-center fixed w-screen h-screen bg-black bg-opacity-60 backdrop-blur-sm"
        style="opacity: 0.013888888888888892;"
      >
        <div
          aria-describedby=":r0:-description"
          aria-labelledby=":r0:-label"
          class="relative bg-white m-4 rounded-lg shadow-2xl text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed w-full md:w-3/5 lg:w-2/5 2xl:w-1/4 min-w-[80%] md:min-w-[60%] lg:min-w-[40%] 2xl:min-w-[25%] max-w-[80%] md:max-w-[60%] lg:max-w-[40%] 2xl:max-w-[25%]"
          id=":r1:"
          role="dialog"
          style="opacity: 0.00617283950617284; transform: translateY(-49.69135802469136px) translateZ(0);"
          tabindex="-1"
        >
          <div
            class="flex items-center shrink-0 p-4 text-blue-gray-900 antialiased font-sans text-2xl font-semibold leading-snug"
          >
            Delete Confirmation
          </div>
          <div
            class="relative p-4 text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed"
          >
            Are you sure you want to delete this item?
          </div>
          <div
            class="items-center shrink-0 flex-wrap p-4 text-blue-gray-500 flex justify-end gap-4"
          >
            <button
              class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20"
              type="button"
            >
              common.cancel
            </button>
            <button
              class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-red-500 text-white shadow-md shadow-red-500/20 hover:shadow-lg hover:shadow-red-500/40 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none"
              type="button"
            >
              common.confirm
            </button>
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    data-floating-ui-portal=""
    id=":r7:"
  >
    <div
      style="position: fixed; overflow: auto; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 9999;"
    >
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
      <div
        class="grid place-items-center fixed w-screen h-screen bg-black bg-opacity-60 backdrop-blur-sm"
        style="opacity: 0;"
      >
        <div
          aria-describedby=":r4:-description"
          aria-labelledby=":r4:-label"
          class="relative bg-white m-4 rounded-lg shadow-2xl text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed w-full md:w-3/5 lg:w-2/5 2xl:w-1/4 min-w-[80%] md:min-w-[60%] lg:min-w-[40%] 2xl:min-w-[25%] max-w-[80%] md:max-w-[60%] lg:max-w-[40%] 2xl:max-w-[25%]"
          id=":r5:"
          role="dialog"
          style="opacity: 0; transform: translateY(-50px) translateZ(0);"
          tabindex="-1"
        >
          <div
            class="flex items-center shrink-0 p-4 text-blue-gray-900 antialiased font-sans text-2xl font-semibold leading-snug"
          >
            Delete Confirmation
          </div>
          <div
            class="relative p-4 text-blue-gray-500 antialiased font-sans text-base font-light leading-relaxed"
          >
            Are you sure you want to delete this item?
          </div>
          <div
            class="items-center shrink-0 flex-wrap p-4 text-blue-gray-500 flex justify-end gap-4"
          >
            <button
              class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg text-gray-900 hover:bg-gray-900/10 active:bg-gray-900/20"
              type="button"
            >
              common.cancel
            </button>
            <button
              class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-red-500 text-white shadow-md shadow-red-500/20 hover:shadow-lg hover:shadow-red-500/40 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none"
              type="button"
            >
              common.confirm
            </button>
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        data-aria-hidden="true"
        data-floating-ui-focus-guard=""
        data-type="inside"
        role="button"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;
